spec/strava/api/client/endpoints/activities/activity_spec.rb

```diff
@@ -278,9 +278,50 @@
 
     expect(activity.available_zones).to eq %w[heartrate pace]
   end
+
   it 'returns activity by id' do
     activity = client.activity(1_946_417_534)
     expect(activity).to be_a Strava::Models::Activity
     expect(activity.id).to eq 1_946_417_534
   end
+
+  describe 'serialization to JSON' do
```

Comment:
I think your tests should also reproduce the original problem of "stack level too deep".